Doc/library/zipfile.rst

```diff
@@ -120,6 +120,29 @@ The module defines the following items:
       methods, and may either refuse to process the ZIP file altogether,
       or fail to extract individual files.
 
+.. data:: PERMS_PRESERVE_NONE
```

Comment:
If these were turned into an enum, it could be:
```python
class PreserveMode(enum.Enum):
    NONE = auto()
    SAFE = auto()
    ALL = auto()
```
which would look ok I think (name not too long and not too generic/short/cryptic):
```python
thing = myzip.extract(member, preserve_permissions=PreserveMode.SAFE)
```